book/src/dev/rfcs/0012-value-pools.md

```diff
@@ -0,0 +1,437 @@
+- Feature Name: value_pools
+- Start Date: 2021-06-29
+- Design PR: [ZcashFoundation/zebra#0000](https://github.com/ZcashFoundation/zebra/pull/0000)
+- Zebra Issue: [ZcashFoundation/zebra#2152](https://github.com/ZcashFoundation/zebra/issues/2152)
+
+# Summary
+[summary]: #summary
+
+This document describes how to verify the Zcash chain and transaction value pools in Zebra.
+
+# Motivation
+[motivation]: #motivation
+
+In the Zcash protocol there are consensus rules that:
+    - prohibit negative chain value pools [ZIP-209], and
+    - restrict the creation of new money to a specific number of coins in each coinbase transaction. [Spec Section 3.4](https://zips.z.cash/protocol/protocol.pdf#transactions)
+
+These rules make sure that a fixed amount of Zcash is created by each block, even if there are vulnerabilities in some shielded pools.
+
+Checking the coins created by coinbase transactions is out of scope.
+
+[ZIP-209]: https://zips.z.cash/zip-0209
+
+# Definitions
+[definitions]: #definitions
+
+- `value balance` - the total value represented by a subset of the blockchain.
+- `transparent value balance` - the sum of the outputs spent by all transparent `tx_in` fields, minus the sum of all `tx_out` fields.
+- `sprout value balance` - the sum of all sprout `vpub_old` fields, minus the sum of all `vpub_new` fields.
+- `sapling value balance` - the negation of the sum of all `valueBalanceSapling` fields.
+- `orchard value balance` - the negation of the sum of all `valueBalanceOrchard` fields.
+- `transaction value pool balance` - the sum of all the value balances in each transaction.
```

Comment:
I think this is the source of my confusion. Is this the sum of all e.g. `orchard value balance`s or of all `valueBalanceOrchard`s? I think it's the latter. (If so, I suggest making this clearer)

For example, a tx with 10 ZEC transparent inputs, 4 ZEC transparent outputs, and 5 ZEC in Orchard outputs:

- valueBalanceOrchard: -5
- orchard_value_balance: 5
- transparent_value_balance: 6 (as defined above)

So

- transaction_value_pool_balance: transparent_value_balance + valueBalanceOrchard = 6 + -5  = 1

An alternative would be to define `transaction value pool balance` as the sum of `transparent value balance` minus the sum of the shielded balances, then

- transaction_value_pool_balance: transparent_value_balance - orchard_value_balance = 6 - 5 = 1

Reply:
https://github.com/ZcashFoundation/zebra/pull/2430/commits/6cff5ee6aaaf84c0095a65de475d9301e8ba593d